sycl/doc/extensions/proposed/sycl_ext_oneapi_queue_status_query.asciidoc

```diff
@@ -0,0 +1,125 @@
+= sycl_ext_intel_queue_status_querry.asciidoc
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2022-2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 4 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+== Overview
+
+This extension adds `sycl::queue::size()`, `sycl::queue::empty()`, `sycl::queue::get_wait_list()` member functions.
+
+- `sycl::queue::size()` and `sycl::queue::empty` allow users to query the number of enqueued commands to a queue.
+- `sycl::queue::get_wait_list()` is the queue equivalent of `event::get_wait_list()` 
+
+== Specification
+
+=== Feature test macro
+
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_QUEUE_STATUS_QUERY` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+
+=== Signatures and Semantics 
+
+This extension adds the following new member functions to the SYCL `queue` class.
+```
+namespace sycl {
+namespace queue {
+ bool empty() const;
+ size_t size() const;
+ std::vector<event> get_wait_list() const;
+}}
```

Comment:
```suggestion
namespace sycl {

class queue {
 bool ext_oneapi_empty() const;
 size_t ext_oneapi_size() const;
 std::vector<event> ext_oneapi_get_wait_list() const;
};

} // namespace sycl
```

These functions are member of `class queue`, not free functions in `namespace queue`.  Also, extended member functions should follow the [extension guidelines](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_names_for_extensions_to_existing_classes_or_enumerations) and start with the prefix `ext_oneapi_`.